pkg/types/types.go

```diff
@@ -231,7 +239,21 @@ func (s byStepTime) Swap(i, j int) {
 }
 
 func (s byStepTime) Less(i, j int) bool {
-	return s[i].StepTime < s[j].StepTime
+	// no deterministic backend sorting
+	if s[i].StepTime != s[j].StepTime || s[i].backendAddr == "" {
+		return s[i].StepTime < s[j].StepTime
+	}
+
+	if s[i].backendWeight < s[j].backendWeight {
```

Comment:
What's the advantage of comparing metricName+address hashes as opposed to just checking addresses as happens in `// dealing with hash collision`?

Reply:
the downside of just sorting by server address is that we might potentially always return metrics in a dc/az due to it's hostnames or ip addresses.
I think it's better to uniformly retrieve metrics from different replicas but still stick to the same server in every api calls.

Reply:
I see. I think given it's not obvious, maybe worth adding a comment.

Would it be sufficient to hash address w/ metric name to achieve the same result?

Reply:
> I see. I think given it's not obvious, maybe worth adding a comment.

yep, sounds good.

> Would it be sufficient to hash address w/o the metric name to achieve the same result?

then it's just the same as sorting backend address?

Reply:
Adds the comment here: https://github.com/bookingcom/carbonapi/pull/340/files#diff-9b4292c30cc53778b4e267af519527527024f6c158979cb0a0a55ca0192651a0R162-R165

Reply:
> then it's just the same as sorting backend address?
true, nvm